plugin/advancedsubscription/ajax/advsub.ajax.php

```diff
@@ -0,0 +1,248 @@
+<?php
+/* For licensing terms, see /license.txt */
+/**
+ * Script to receipt request to subscribe and confirmation action to queue
+ * @author Daniel Alejandro Barreto Alva <daniel.barreto@beeznest.com>
+ * @package chamilo.plugin.advancedsubscription
+ */
+
+/**
+ * Init
+ */
+require_once __DIR__ . '/../config.php';
+
+$plugin = AdvancedSubscriptionPlugin::create();
+// Get validation hash
+$hash = Security::remove_XSS($_REQUEST['v']);
+// Get data from request (GET or POST)
+$data['a'] = Security::remove_XSS($_REQUEST['a']);
+$data['s'] = intval($_REQUEST['s']);
+$data['current_user_id'] = intval($_REQUEST['current_user_id']);
+$data['u'] = intval($_REQUEST['u']);
+$data['q'] = intval($_REQUEST['q']);
+$data['e'] = intval($_REQUEST['e']);
+$data['is_connected'] = isset($_REQUEST['is_connected']) ? boolval($_REQUEST['is_connected']) : false;
+$data['profile_completed'] = isset($_REQUEST['profile_completed']) ? floatval($_REQUEST['profile_completed']) : 0;
+// Init result array
+$result = array('error' => true, 'errorMessage' => get_lang('ThereWasAnError'));
+// Check if data is valid or is for start subscription
+$verified = $plugin->checkHash($data, $hash) || $data['a'] == 'subscribe';
+if ($verified) {
+    switch($data['a']) {
+        case 'check': // Check minimum requirements
+            try {
+                $res = AdvancedSubscriptionPlugin::create()->isAbleToRequest($data['u'], $data);
+                if ($res) {
+                    $result['error'] = false;
+                    $result['errorMessage'] = 'No error';
+                    $result['pass'] = true;
+                } else {
+                    $result['errorMessage'] = 'User can not be subscribed';
+                    $result['pass'] = false;
+                }
+            } catch (\Exception $e) {
+                $result['errorMessage'] = $e->getMessage();
+            }
+            break;
+        case 'subscribe': // Subscription
+            // Start subscription to queue
+            $res = AdvancedSubscriptionPlugin::create()->startSubscription($data['u'], $data['s'], $data);
+            // Check if queue subscription was successful
+            if ($res === true) {
+                // Prepare data
+                // Get session data
+                $sessionArray = api_get_session_info($data['s']);
+                $extraSession = new ExtraFieldValue('session');
+                $var = $extraSession->get_values_by_handler_and_field_variable($data['s'], 'description');
```

Comment:
Given the number of calls to get_values_by_handler_and_field_variable(), I think it would make more sense to get *all* the extrafields for this session first, and *then* check each of them separately (same thing further down this switch())